### PR TITLE
Fix mac monterey

### LINKED ISSUE #1030: Fail to write files on mac monterey
[20211220-135633585.txt](https://github.com/bluerobotics/ping-viewer/files/7788832/20211220-135633585.txt)

### PR DESCRIPTION
**Pull request by patrickelectric**
- Fix #1030

### CODE REVIEW

**Comment by patrickelectric** [comment]
Tested and working.

**Comment by patrickelectric** [comment]
ping @ES-Alexander, @Williangalvani

File: src/link/filelink.cpp [hunk 1]
```diff
@@ -184,10 +179,22 @@ bool FileLink::isOpen()
...
     // To return at least a good answer, we do check the path to see if it's writable
-    return (QFileInfo(QFileInfo(_file).canonicalPath()).isWritable() && _openModeFlag == QIODevice::WriteOnly)
+    return (isWritable() && _openModeFlag == QIODevice::WriteOnly)
         || _file.isReadable(); // If file is readable it's already opened and working
 };
 
+bool FileLink::isWritable()
+{
+    // We should use the absoluteBasePath to make sure that the path is writable
+    // Using the complete path with the file name may result in problems (E.g: Mac)
```
Comment (Williangalvani) [inline src/link/filelink.cpp:189]:
```suggestion
    // Using the relative path with the file name may result in problems (E.g: Mac)
```

Comment (Williangalvani) [reply src/link/filelink.cpp:189]: oh, nevermind. I read it wrong

**Review by Williangalvani — APPROVED** [review]
I can't test this, but the code looks good.

**Pull request merged**